lib/api/core/plugins/pluginContext.js

```diff
@@ -157,48 +157,37 @@ class PluginContext {
  * @param {boolean} [overloadProtect]
  * @param {Function} [callback]
  */
-function execute (kuzzle, request, overloadProtect = true, callback = null) {
-  let resolve = null,
-    reject = null, 
-    deferred = null,
-    _callback = callback,
-    _overloadProtect = overloadProtect;
-
-  let error = null;
-  const overloadProtectType = typeof _overloadProtect;
-
-  if (overloadProtectType === 'function') {
-    _callback = _overloadProtect;
-    _overloadProtect = true;
-  }
-  else if (overloadProtectType !== 'boolean') {
-    error = new PluginImplementationError('Invalid argument: the overload protection flag must be a boolean');
-  }
+function execute (kuzzle, request, callback) {
+  let
+    error,
+    resolve,
+    reject,
+    deferred;
 
-  if (!_callback) {
+  if (!callback) {
     deferred = new Bluebird((res, rej) => {
       resolve = res;
       reject = rej;
     });
   }
 
-  if (!error && (!request || !(request instanceof Request))) {
+  if (!request || !(request instanceof Request)) {
```

Comment:
Nitpicking: it seems `error` can now only set by next line...
So lines 174->185 can be simplified like that:
```javascript
if (!request || !(request instanceof Request)) {
  const error = new PluginImplementationError('Invalid argument: a Request object must be supplied');
  if (callback) {
    return callback(error);
  }
  reject(error);
  return deferred;
}
```

Reply:
done